internal/gogrep/gen_operations.go

```diff
@@ -0,0 +1,311 @@
+	{name: "IfInitStmt", tag: "IfStmt", args: "init cond block", example: "if init; cond {}"},
+	{name: "IfElseStmt", tag: "IfStmt", args: "cond block else", example: "if cond {} else ..."},
+	{name: "IfInitElseStmt", tag: "IfStmt", args: "init cond block else", example: "if init; cond {} else ..."},
+	{name: "IfNamedOptStmt", tag: "IfStmt", args: "block", valueIndex: "strings | wildcard name", example: "if $*x {}"},
+	{name: "IfNamedOptElseStmt", tag: "IfStmt", args: "block else", valueIndex: "strings | wildcard name", example: "if $*x {} else ..."},
+
+	{name: "SwitchStmt", tag: "SwitchStmt", args: "body...", example: "switch {}"},
+	{name: "SwitchTagStmt", tag: "SwitchStmt", args: "tag body...", example: "switch tag {}"},
+	{name: "SwitchInitStmt", tag: "SwitchStmt", args: "init body...", example: "switch init; {}"},
+	{name: "SwitchInitTagStmt", tag: "SwitchStmt", args: "init tag body...", example: "switch init; tag {}"},
+
+	{name: "SelectStmt", tag: "SelectStmt", args: "body..."},
+
+	{name: "TypeSwitchStmt", tag: "TypeSwitchStmt", args: "x block", example: "switch x.(type) {}"},
+	{name: "TypeSwitchInitStmt", tag: "TypeSwitchStmt", args: "init x block", example: "switch init; x.(type) {}"},
+
+	{name: "CaseClause", tag: "CaseClause", args: "values... body..."},
+	{name: "DefaultCaseClause", tag: "CaseClause", args: "body..."},
+
+	{name: "CommClause", tag: "CommClause", args: "comm body..."},
+	{name: "DefaultCommClause", tag: "CommClause", args: "body..."},
+
+	{name: "ForStmt", tag: "ForStmt", args: "blocl", example: "for {}"},
+	{name: "ForPostStmt", tag: "ForStmt", args: "post block", example: "for ; ; post {}"},
+	{name: "ForCondStmt", tag: "ForStmt", args: "cond block", example: "for ; cond; {}"},
+	{name: "ForCondPostStmt", tag: "ForStmt", args: "cond post block", example: "for ; cond; post {}"},
+	{name: "ForInitStmt", tag: "ForStmt", args: "init block", example: "for init; ; {}"},
+	{name: "ForInitPostStmt", tag: "ForStmt", args: "init post block", example: "for init; ; post {}"},
+	{name: "ForInitCondStmt", tag: "ForStmt", args: "init cond block", example: "for init; cond; {}"},
+	{name: "ForInitCondPostStmt", tag: "ForStmt", args: "init cond post block", example: "for init; cond; post {}"},
+
+	{name: "RangeStmt", tag: "RangeStmt", args: "x block", example: "for range x {}"},
+	{name: "RangeKeyStmt", tag: "RangeStmt", args: "key x block", value: "token.Token | ':=' or '='", example: "for key := range x {}"},
+	{name: "RangeKeyValueStmt", tag: "RangeStmt", args: "key value x block", value: "token.Token | ':=' or '='", example: "for key, value := range x {}"},
+
+	{name: "FieldList", args: "fields..."},
+	{name: "UnnamedField", args: "typ", example: "type"},
+	{name: "SimpleField", args: "typ", valueIndex: "strings | field name", example: "name type"},
+	{name: "Field", args: "name typ", example: "$name type"},
+	{name: "MultiField", args: "names... typ", example: "name1, name2 type"},
```

Comment:
no `tag` for this 5 ?

Reply:
Yep. Instructions that can't be mapped to AST nodes have no tag.
They're usually a part of some other sequence that can be mapped to AST.